clients/unshaded/src/main/java/tachyon/client/block/RemoteBlockInStream.java

```diff
@@ -30,6 +33,13 @@
 public final class RemoteBlockInStream extends BufferedBlockInStream {
   /** The address of the worker to read the data from. */
   private final InetSocketAddress mLocation;
+  /** The returned lock id after acquiring the block lock. */
+  private final Long mLockId;
+
+  /** Client to communicate with the remote worker. */
+  private final WorkerClient mWorkerClient;
```

Comment:
Currently we need to have the same client do it because the session id needs to match. However, clients probably shouldn't have the state and it should be part of the stream (not related to this PR).